static/js/components/Reminders.jsx

```diff
@@ -0,0 +1,449 @@
+  buttonIcon: {
+    maxWidth: "1.2rem",
+  },
+}));
+
+// Tweak responsive styling
+const getMuiTheme = (theme) =>
+  createTheme(
+    adaptV4Theme({
+      palette: theme.palette,
+      overrides: {
+        MUIDataTablePagination: {
+          toolbar: {
+            flexFlow: "row wrap",
+            justifyContent: "flex-end",
+            padding: "0.5rem 1rem 0",
+            [theme.breakpoints.up("sm")]: {
+              // Cancel out small screen styling and replace
+              padding: "0px",
+              paddingRight: "2px",
+              flexFlow: "row nowrap",
+            },
+          },
+          tableCellContainer: {
+            padding: "1rem",
+          },
+          selectRoot: {
+            marginRight: "0.5rem",
+            [theme.breakpoints.up("sm")]: {
+              marginLeft: "0",
+              marginRight: "2rem",
+            },
+          },
+        },
+      },
+    })
+  );
+
+const dialogTitleStyles = (theme) => ({
+  root: {
+    margin: 0,
+    padding: theme.spacing(2),
+  },
+  title: {
+    marginRight: theme.spacing(2),
+  },
+  closeButton: {
+    position: "absolute",
+    right: theme.spacing(1),
+    top: theme.spacing(1),
+    color: grey[500],
+  },
+});
+
+const DialogTitle = withStyles(dialogTitleStyles)(
+  ({ children, classes, onClose }) => (
+    <MuiDialogTitle className={classes.root}>
+      <Typography variant="h6" className={classes.title}>
+        {children}
+      </Typography>
+      {onClose ? (
+        <IconButton
+          aria-label="close"
+          className={classes.closeButton}
+          onClick={onClose}
+        >
+          <Close />
+        </IconButton>
+      ) : null}
+    </MuiDialogTitle>
+  )
+);
+
+const NewReminder = ({ resourceId, resourceType, handleClose }) => {
+  const dispatch = useDispatch();
+  const defaultDate = dayjs().utc().format("YYYY-MM-DDTHH:mm:ssZ");
+
+  const handleSubmit = ({ formData }) => {
+    formData.next_reminder = formData.next_reminder
+      .replace("+00:00", "")
+      .replace(".000Z", "");
+    dispatch(Actions.submitReminder(resourceId, resourceType, formData)).then(
+      (response) => {
+        if (response.status === "success") {
+          dispatch(showNotification("Reminder created"));
+          handleClose();
+        } else {
+          dispatch(showNotification("Error creating reminder", "error"));
+        }
+      }
+    );
+  };
+
+  function validate(formData, errors) {
+    if (formData.text === "") {
+      errors.text = "Reminder text is required";
+    }
+    if (defaultDate > formData.next_reminder) {
+      errors.next_reminder.addError("Next reminder date can't be in the past");
+    }
+    if (formData.number_of_reminders <= 0) {
+      errors.number_of_reminders.addError(
+        "Number of reminders must be greater than 0"
+      );
+    }
+    if (formData.reminder_delay <= 0) {
+      errors.reminder_delay.addError(
+        "Reminder delay must be greater than 0 day"
+      );
+    }
```

Comment:
If the next reminder is in the future, and `number_of_reminders=1` then this field is actually irrelevant.

Reply:
What do you mean by that? That we can calculate the delay based on the difference between the current time and the next reminder? To me, here's how I understood the feature: the user chooses the next_reminder(that is the date of the first time it will be triggered), the number of reminder, and the delay in between the reminders. So I don't see why this field is irrelevant.

Reply:
If I set a reminder with `number=1` and set it to five minutes in the future. The 1 day delay never gets called because after those five minutes the reminder is finished. 
I guess it doesn't matter much for this form, because there's a reasonable default in the `delay` field so if the user ignores this it doesn't hurt.